extra/modules/fiftyone-devicedetection/src/main/java/org/prebid/server/hooks/modules/fiftyone/devicedetection/v1/hooks/FiftyOneDeviceDetectionRawAuctionRequestHook.java

```diff
@@ -0,0 +1,156 @@
+                        .status(InvocationStatus.success)
+                        .action(InvocationAction.update)
+                        .payloadUpdate(freshPayload -> updatePayload(freshPayload, moduleContext.collectedEvidence()))
+                        .moduleContext(moduleContext)
+                        .build()
+        );
+    }
+
+    private boolean isAccountAllowed(AuctionInvocationContext invocationContext) {
+        final List<String> allowList = ObjectUtil.getIfNotNull(accountFilter, AccountFilter::getAllowList);
+        if (CollectionUtils.isEmpty(allowList)) {
+            return true;
+        }
+        return Optional.ofNullable(invocationContext)
+                .map(AuctionInvocationContext::auctionContext)
+                .map(AuctionContext::getAccount)
+                .map(Account::getId)
+                .filter(StringUtils::isNotBlank)
+                .map(allowList::contains)
+                .orElse(false);
+    }
+
+    private ModuleContext addEvidenceToContext(ModuleContext moduleContext,
+                                               Consumer<CollectedEvidence.CollectedEvidenceBuilder> evidenceInjector) {
+        final CollectedEvidence.CollectedEvidenceBuilder evidenceBuilder = Optional.ofNullable(moduleContext)
+                .map(ModuleContext::collectedEvidence)
+                .map(CollectedEvidence::toBuilder)
+                .orElseGet(CollectedEvidence::builder);
+
+        evidenceInjector.accept(evidenceBuilder);
+
+        return Optional.ofNullable(moduleContext)
+                .map(ModuleContext::toBuilder)
+                .orElseGet(ModuleContext::builder)
+                .collectedEvidence(evidenceBuilder.build())
+                .build();
+    }
+
+    private void collectEvidence(CollectedEvidence.CollectedEvidenceBuilder evidenceBuilder, BidRequest bidRequest) {
+        final Device device = ObjectUtil.getIfNotNull(bidRequest, BidRequest::getDevice);
+        if (device == null) {
+            return;
+        }
+        final String ua = device.getUa();
+        if (ua != null) {
+            evidenceBuilder.deviceUA(ua);
+        }
+        final UserAgent sua = device.getSua();
+        if (sua != null) {
+            evidenceBuilder.secureHeaders(convertSecureHeaders(sua));
+        }
+    }
+
+    private Map<String, String> convertSecureHeaders(UserAgent userAgent) {
+        return SecureHeadersRetriever.retrieveFrom(userAgent);
+    }
+
+    private AuctionRequestPayload updatePayload(AuctionRequestPayload existingPayload,
+                                                CollectedEvidence collectedEvidence) {
+        final BidRequest currentRequest = existingPayload.bidRequest();
+        final BidRequest patchedRequest = enrichDevice(currentRequest, collectedEvidence);
+        if (patchedRequest == null || patchedRequest == currentRequest) {
+            return existingPayload;
+        }
+        return AuctionRequestPayloadImpl.of(patchedRequest);
```

Comment:
does this comparison even matters?
I believe it could be just as the following
```
return patchedRequest == null ? existingPayload : AuctionRequestPayloadImpl.of(patchedRequest);
```

Reply:
:heavy_check_mark: Applied